src/main/java/frc/robot/command/AutoAlignCommand.java

```diff
@@ -0,0 +1,217 @@
+    // Determine which auto-align button is pressed; assume only one is held.
+    if (controller.getAutoAlignCoralLeft()) {
+      selectedMode = AutoAlignMode.CORAL_LEFT;
+    } else if (controller.getAutoAlignCoralRight()) {
+      selectedMode = AutoAlignMode.CORAL_RIGHT;
+    } else if (controller.getAutoAlignAlgaeMiddle()) {
+      selectedMode = AutoAlignMode.ALGAE_MIDDLE;
+    } else {
+      // No button is held; cancel command.
+      cancel();
+      return;
+    }
+
+    // Get current robot pose from PositioningSubsystem
+    Pose2d currentPose = positioning.getRobotPose();
+
+    // Find the closest master node using Euclidean distance
+    Pose2d closestMaster = null;
+    double minDistance = Double.POSITIVE_INFINITY;
+    for (Pose2d master : masterNodes) {
+      double distance = currentPose.getTranslation().getDistance(master.getTranslation());
+      if (distance < minDistance) {
+        minDistance = distance;
+        closestMaster = master;
+      }
+    }
+
+    if (closestMaster == null) {
+      // Safety: if no master node is found, cancel command.
+      cancel();
+      return;
+    }
+
+    // Generate the absolute trajectory from the current pose to the closest master node.
+    absoluteTrajectory = PathPlanner.generateTrajectory(
+            currentPose,
+            List.of(), // no intermediate waypoints
+            closestMaster,
+            trajConfig
+    );
+
+    // Determine the desired sub node offset based on the selected mode.
+    Translation2d subOffset = subNodeOffsets.get(selectedMode);
+    // Compute final target for the second phase:
+    Pose2d subTarget = new Pose2d(closestMaster.getTranslation().plus(subOffset), closestMaster.getRotation());
+
+    // Generate a relative trajectory from the master node to the sub node.
+    // This trajectory is only translational; the rotation remains as the master node's rotation.
+    relativeTrajectory = PathPlanner.generateTrajectory(
+            closestMaster,
+            List.of(), // no intermediate waypoints
+            subTarget,
+            trajConfig
+    );
+
+    // Create trajectory following commands.
+    // Here we use PPSwerveControllerCommand (assumed to be configured similarly to your teleop drive controller)
+    Command absoluteFollow = new PPSwerveControllerCommand(
+            absoluteTrajectory,
+            positioning::getPose,
+            drive.kinematics,
+            drive.getDriveController(),
+            drive::driveSwerveStates,
+            drive
+    );
+
+    Command relativeFollow = new PPSwerveControllerCommand(
+            relativeTrajectory,
+            // For the relative trajectory, we use a lambda that returns the master node pose
+            // (the trajectory is relative; no additional rotation correction is needed)
+            () -> closestMaster,
+            drive.kinematics,
+            drive.getDriveController(),
+            drive::driveSwerveStates,
+            drive
+    );
```

Comment:
We already have a method to create a command that follows a PathPlanner path. Refer to [PathPlannerCommands#getFollowPathCommand](https://github.com/momentumfrc/2025Reefscape/blob/main/src/main/java/frc/robot/utils/PathPlannerCommands.java)